src/components/Visualiser/utils/node-calculator.ts

```diff
@@ -3,9 +3,12 @@ import { isNode, Node } from 'react-flow-renderer';
 const groupNodesByColumn = (elements: Node[]) => {
   return elements.reduce((elementsGrouped: any, element: Node) => {
     if (isNode(element) && element.__rf) {
+      const assignElementsGrouped = () => {
+        elementsGrouped[element?.data.groupId] = [element];
+      };
       return {
         ...elementsGrouped,
-        [element.data.columnToRenderIn]: elementsGrouped[element?.data.columnToRenderIn] ? elementsGrouped[element?.data.columnToRenderIn].concat([element]) : (elementsGrouped[element?.data.groupId] = [element]),
+        [element.data.columnToRenderIn]: elementsGrouped[element?.data.columnToRenderIn] ? elementsGrouped[element?.data.columnToRenderIn].concat([element]) : assignElementsGrouped(),
```

Comment:
Can we make it in different way, removing ternary `? :` operator and base logic on if/else?

Reply:
Should I create a separate function that handles the assignment with if-else?

Reply:
Nope, make it in the `reduce` callback.

Reply:
Would this suffice?

```
if (isNode(element) && element.__rf) {
      if (elementsGrouped[element?.data.columnToRenderIn]) {
        return {
          ...elementsGrouped,
          [element.data.columnToRenderIn]: elementsGrouped[element?.data.columnToRenderIn].concat([element])};
      }
      
      return {
        ...elementsGrouped,
        [element.data.columnToRenderIn]: (elementsGrouped[element?.data.groupId] = [element]),
      };
    }
```